File: algorithms/[704]二分查找/solution.ts
// topics = ["二分查找"]

function search(nums: number[], target: number): number {
  let i = 0;
  let j = nums.length - 1;

  while (i <= j) {
    const mid = Math.floor((i + j) / 2);
    if (nums[mid] == target) {
      return mid;
    }
    if (nums[mid] < target) {
      i = mid + 1;
    } else {
      j = mid - 1;
    }
  }

  return -1;
}

File: algorithms/[704]二分查找/solution.py
# topics = ["二分查找"]

from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums)

        while left < right:
            mid = (left + right) // 2
            if target == nums[mid]:
                return mid
            if nums[mid] < target:
                left = mid + 1
            else:
                right = mid

        return -1
